File: JetStreamMongo/Models/ServiceAuftrag.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace JetStreamMongo.Models
{
    /// <summary>
    /// Represents a service order within the system.
    /// </summary>
    public class ServiceAuftrag
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("Name")]
        public string Name { get; set; }

        [BsonElement("Email")]
        public string Email { get; set; }

        [BsonElement("Phone")]
        public string Phone { get; set; }

        [BsonElement("Priority")]
        public string Priority { get; set; }

        [BsonElement("Service")]
        public string Service { get; set; }

        
        public DateTime CreateDate { get; set; }

        
        public DateTime PickupDate { get; set; }
    }

}

File: mongoScriptProj/scripts/ServiceAuftrag.js
// serviceAuftragSetup.js
const jet = db.getSiblingDB('jetstream');

jet.createCollection('ServiceAuftrag', {
    validator: {
       $jsonSchema: {
           bsonType: "object",
           required: [ "Name", "Email", "Phone", "Priority", "Service", "CreateDate", "PickupDate" ],
           properties: {
               _id: {
                   bsonType: "objectId",
                   description: "muss ein gültiger ObjectId sein"
               },
               Name: {
                   bsonType: "string",
                   description: "muss vorhanden und eine Zeichenkette sein"
               },
               Email: {
                   bsonType: "string",
                   pattern: "^.+@.+\..+$",
                   description: "muss eine gültige E-Mail-Adresse sein"
               },
               Phone: {
                   bsonType: "string",
                   description: "muss vorhanden und eine Zeichenkette sein"
               },
               Priority: {
                   bsonType: "string",
                   description: "muss vorhanden und eine Zeichenkette sein"
               },
               Service: {
                   bsonType: "string",
                   description: "muss vorhanden und eine Zeichenkette sein"
               },
               CreateDate: {
                   bsonType: "date",
                   description: "muss vorhanden und ein Datum sein"
               },
               PickupDate: {
                   bsonType: "date",
                   description: "muss vorhanden und ein Datum sein"
               }
           }
       }
    }
});

jet.ServiceAuftrag.createIndex({ Service: 1 }, { unique: false });


